### PR TITLE
Prepare Mainnet for tx soft fork

### PR DESCRIPTION
**Pull request by Yostra**

### CODE REVIEW

**Review by hoffmang9 — APPROVED** [review]
+1

File: src/consensus/block_body_validation.py [hunk 1]
```diff
@@ -155,231 +152,247 @@ async def validate_block_body(
...
-    if block.transactions_generator is not None:
-        # Get List of names removed, puzzles hashes for removed coins and conditions crated
-        if cached_cost_result is not None:
-            result: Optional[CostResult] = cached_cost_result
+    if constants.NETWORK == NetworkType.MAINNET:
+        if block.transactions_generator is not None:
+            if len(bytes(block.transactions_generator)) > constants.MAX_GENERATOR_SIZE:
+                return Err.PRE_SOFT_FORK_MAX_GENERATOR_SIZE, None
+            else:
+                return None, None
```
Comment (mariano54) [inline src/consensus/block_body_validation.py:160]: We should return an error here instead of None

File: src/consensus/block_body_validation.py [hunk 2]
```diff
@@ -155,231 +152,247 @@ async def validate_block_body(
...
+            if rem in additions_dic:
+                # Ephemeral coin
+                rem_coin: Coin = additions_dic[rem]
+                new_unspent: CoinRecord = CoinRecord(
+                    rem_coin,
+                    height,
                     uint32(0),
                     False,
-                    (rem in coinbases_since_fork),
+                    False,
```
Comment (mariano54) [inline src/consensus/block_body_validation.py:294]: Why this change?

File: src/simulator/start_simulator.py [hunk 3]
```diff
@@ -20,16 +20,14 @@
 SERVICE_NAME = "full_node"
 
 
-def service_kwargs_for_full_node_simulator(
-    root_path: Path,
-    config: Dict,
-    bt: BlockTools,
-) -> Dict:
+def service_kwargs_for_full_node_simulator(root_path: Path, config: Dict, bt: BlockTools, constants_dic=None) -> Dict:
```
Comment (mariano54) [inline src/simulator/start_simulator.py:23]: This doesn't seem great. Can we instead replace the field in block tools constants?

File: src/consensus/block_body_validation.py (hunk 1 above)
Comment (Yostra) [reply src/consensus/block_body_validation.py:160]: in case of else: ? - there's no error

File: src/consensus/block_body_validation.py (hunk 2 above)
Comment (Yostra) [reply src/consensus/block_body_validation.py:294]: hmm maybe bad rebase, I'll check

File: src/simulator/start_simulator.py (hunk 3 above)
Comment (Yostra) [reply src/simulator/start_simulator.py]: fixed, moved to setup nodes / blocktools
